packages/seal-mongodb-adapter/src/MongoDBAdapter.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the CMS-IG SEAL project.
+ *
+ * (c) Alexander Schranz <alexander@sulu.io>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CmsIg\Seal\Adapter\MongoDB;
+
+use CmsIg\Seal\Adapter\AdapterInterface;
+use CmsIg\Seal\Adapter\IndexerInterface;
+use CmsIg\Seal\Adapter\SchemaManagerInterface;
+use CmsIg\Seal\Adapter\SearcherInterface;
+
+final class MongoDBAdapter implements AdapterInterface
+{
+    private readonly SchemaManagerInterface $schemaManager;
+
+    private readonly IndexerInterface $indexer;
+
+    private readonly SearcherInterface $searcher;
+
+    public function __construct(
+        ClientWrapper $client,
```

Comment:
You can inject the `MongoDB\Database` object instead; instantiation does not open the connection.

Reply:
I like the idea, if I can do all over the database I think that would be the best option.